api/src/services/authorization.js

```diff
@@ -0,0 +1,225 @@
+let db = require('../db-pouch'),
+    auth = require('../auth'),
+    _ = require('underscore'),
+    config = require('../config'),
+    viewMapUtils = require('@shared-libs/view-map-utils'),
+    tombstoneUtils = require('@shared-libs/tombstone-utils')(Promise, db.medic);
+
+const ALL_KEY = '_all', // key in the docs_by_replication_key view for records everyone can access
+      UNASSIGNED_KEY = '_unassigned'; // key in the docs_by_replication_key view for unassigned records
+
+const getDepth = (userCtx) => {
+  if (!userCtx.roles || !userCtx.roles.length) {
+    return -1;
+  }
+
+  const settings = config.get('replication_depth');
+  if (!settings) {
+    return -1;
+  }
+  let depth = -1;
+  userCtx.roles.forEach(function(role) {
+    // find the role with the deepest depth
+    const setting = _.findWhere(settings, { role: role });
+    const settingDepth = setting && parseInt(setting.depth, 10);
+    if (!isNaN(settingDepth) && settingDepth > depth) {
+      depth = settingDepth;
+    }
+  });
+  return depth;
+};
+
+const hasAccessToUnassignedDocs = (userCtx) => {
+  return config.get('district_admins_access_unallocated_messages') &&
+         auth.hasAllPermissions(userCtx, 'can_view_unallocated_data_records');
+};
+
+const exclude = (array, ...values) => {
+  if (!Array.isArray(array)) {
+    return;
+  }
+
+  let i = 0;
+  while (i < array.length) {
+    if (values.indexOf(array[i]) !== -1) {
+      array.splice(i, 1);
+    } else {
+      i++;
+    }
+  }
+};
```

Comment:
Why not use array.filter ?

Reply:
I wanted the operation to be mutating. I have since refactored this code, not destructuring the `feed` object, array.filter instead and assigning the new value to the subjectIds.